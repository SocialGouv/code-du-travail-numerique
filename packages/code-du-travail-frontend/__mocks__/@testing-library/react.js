/* eslint-disable import/export */
import { theme } from "@socialgouv/cdtn-ui";
import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
