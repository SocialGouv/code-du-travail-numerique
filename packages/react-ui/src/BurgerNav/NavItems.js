import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Button } from "../Button";
import { box, breakpoints, fonts, spacings } from "../theme.js";

export const BurgerNavButton = styled(Button).attrs(() => ({
  variant: "navLink",
}))`
  position: relative;
  display: flex !important;
  align-items: center;
  height: 100%;
  padding: 0 ${spacings.medium};
  font-weight: normal;
  font-size: ${fonts.sizes.default};
  font-family: "Open Sans", sans-serif;
  border: none;
  @media (max-width: ${breakpoints.tablet}) {
    justify-content: center;
    width: 100%;
    height: 5.4rem;
    padding: 0;
    font-weight: 600;
    font-size: ${fonts.sizes.headings.small};
  }
`;

export const BurgerNavItem = ({ isCurrent, ...props }) =>
  isCurrent ? (
    <BurgerNavCurrent title="Page courante" {...props} />
  ) : (
    <BurgerNavLink {...props} />
  );
BurgerNavItem.propTypes = {
  children: PropTypes.node.isRequired,
  isCurrent: PropTypes.bool,
};
const BurgerNavLink = styled(BurgerNavButton).attrs(() => ({ as: "a" }))`
  text-decoration: none;
`;

const BurgerNavCurrent = styled(BurgerNavButton).attrs(() => ({
  as: "span",
}))`
  cursor: inherit;

  &:after {
    position: absolute;
    bottom: 0;
    left: 50%;
    width: 90%;
    height: 3px;
    background-color: ${({ theme }) => theme.primary};
    border-radius: ${box.borderRadius};
    transform: translateX(-50%);
    content: "";
  }

  @media (max-width: ${breakpoints.tablet}) {
    &:after {
      bottom: auto;
      left: 0;
      width: 3px;
      height: 100%;
      transform: none;
    }
  }
`;
