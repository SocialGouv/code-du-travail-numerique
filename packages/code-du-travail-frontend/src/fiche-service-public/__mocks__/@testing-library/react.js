import React from "react";
import PropTypes from "prop-types";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "@socialgouv/cdtn-ui";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme.colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
