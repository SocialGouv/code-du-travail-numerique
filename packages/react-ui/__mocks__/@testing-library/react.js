import { render } from "@testing-library/react";
import PropTypes from "prop-types";
import React from "react";
import { ThemeProvider } from "styled-components";

import { colors } from "../../src/theme";

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={colors}>
      <>{children}</>
    </ThemeProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
