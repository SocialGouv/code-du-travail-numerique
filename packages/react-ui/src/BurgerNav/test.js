import "jest-styled-components";

import { render } from "@testing-library/react";
import React from "react";

import { BurgerNav } from "./index.js";
import { BurgerNavButton, BurgerNavItem } from "./NavItems.js";

describe("BurgerStyles", () => {
  it("renders", () => {
    const { container } = render(
      <BurgerNav>
        <BurgerNavButton>’Till it goes click</BurgerNavButton>
        <BurgerNavItem
          isCurrent={true}
          href="https://www.youtube.com/watch?v=8xTqP58o1iw&feature=youtu.be&t=20"
        >
          ’Till it goes click
        </BurgerNavItem>
        <BurgerNavItem isCurrent={false}>You said it man</BurgerNavItem>
      </BurgerNav>
    );
    expect(container).toMatchSnapshot();
  });
});
