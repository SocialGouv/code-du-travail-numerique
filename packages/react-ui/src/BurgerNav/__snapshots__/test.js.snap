// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerStyles renders 1`] = `
<div>
  <button
    aria-label="menu principal"
    class="sc-aXZVg fayrVA sc-fqkvVR idbyNV"
  >
    <svg
      aria-hidden="true"
      class="sc-dcJsrY iVptCR"
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M7 11a1 1 0 0 1 1-1h16a1 1 0 1 1 0 2H8a1 1 0 0 1-1-1zm0 5a1 1 0 0 1 1-1h16a1 1 0 1 1 0 2H8a1 1 0 0 1-1-1zm1 4a1 1 0 1 0 0 2h16a1 1 0 1 0 0-2H8z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;
